dhcpv4/nclient4/client.go

```diff
@@ -447,12 +453,19 @@ func (c *Client) Request(ctx context.Context, modifiers ...dhcpv4.Modifier) (off
 		return
 	}
 
-	ack, err = c.SendAndRead(ctx, c.serverAddr, request, nil)
+	ack, err := c.SendAndRead(ctx, c.serverAddr, request, nil)
 	if err != nil {
 		err = fmt.Errorf("got an error while processing the request: %w", err)
 		return
 	}
-
+	lease = &Lease{}
```

Comment:
put the offer in the Lease too? might as well

Reply:
ok, I have added offer in the Lease